tests/draft-next/unevaluatedProperties.json

```diff
@@ -1603,5 +1603,65 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "propertyDependencies with unevaluatedProperties" ,
+        "schema" : {
+            "$schema": "https://json-schema.org/draft/next/schema",
+            "propertyDependencies": {
+                "foo": {
+                    "bar": {
+                        "properties": {
+                            "buz": {}
+                        }
+                    }
+                }
+            },
+            "unevaluatedProperties": false
+        },
+
+        "tests": [
+            {
+                "description": "unevaluatedProperties doesn't consider propertyDependencies" , 
+                "data": {"foo": "bar"},
+                "valid": false
+            },
+            {
+                "description": "unevaluatedProperties doesn't consider inner schema properties",
+                "data": {"foo": "bar", "buz": ""},
+                "valid": false
+            }
```

Comment:
I'm not sure this is actually testing what you want.

With this schema, validation fails because `foo` isn't defined.  Because of that, it doesn't matter whether `unevaluatedProperties` sees `buz` because it's going to fail anyway.

To check whether it sees `buz`, we need to first define `foo`:

```json
{
    "$schema": "https://json-schema.org/draft/next/schema",
    "properties": {
        "foo": { "type": "string" }
    },
    "propertyDependencies": {
        "foo": {
            "bar": {
                "properties": {
                    "buz": {}
                }
            }
        }
    },
    "unevaluatedProperties": false
}
```

with the instance

```json
{"foo": "bar", "buz": ""}
```

This is valid.

Reply:
In this if valid means `valid : true` then i need a clarity in this 
when we test ( i have tested this on hyperjump and jsonschema validator ) 
```json
{
  "$id": "https://json-schema.hyperjump.io/schema",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "properties": {
    "foo": {}
  },
  "dependentSchemas": {
    "foo": {
      "properties": {
        "bar": {}
      }
    }
  },
  "additionalProperties": false
}
```

against `{"foo": "" , "bar": ""}` 
then validation get fails bcz even the innerschema properties (bar: "") are apply , additionalProperties only allow the properties define in the root `properties` 

so i  think this behaviour also same with the propertyDependencies so i think that schema with instance 
`{"foo": "bar", "buz": ""}` is gonna fail ( valid : false ) 

am i correct on this point @gregsdennis ?

Reply:
OK i got it i am wrong in this point , with `unevaluatedProperties` we get this schema valid but with `additionalProperties` we get this schema invalid

Reply:
Not against that instance.  The only properties in your schema are `foo` and `bar`.  `buz` would be invalid for both `additionalProperties` and `unevaluatedProperties`.

However, I see that you changed the inner property from `buz` to `bar`.  Maybe that was a typo?

If so, your schema would be

```json
{
  "$id": "https://json-schema.hyperjump.io/schema",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "properties": {
    "foo": {}
  },
  "dependentSchemas": {
    "foo": {
      "properties": {
        "buz": {}
      }
    }
  },
  "additionalProperties": false
}
```

for which `{"foo": "bar", "buz": ""}` would be invalid; but with `unevaluatedProperties`, it would be valid.